lib/shortcode/presenter.rb

```diff
@@ -7,9 +7,7 @@ def self.register(configuration, presenter)
 
   def self.validate(presenter)
     raise ArgumentError, "The presenter must define the class method #for" unless presenter.respond_to?(:for)
-    unless presenter.private_instance_methods(false).include?(:initialize)
-      raise ArgumentError, "The presenter must define an initialize method"
-    end
+    raise ArgumentError, "The presenter must define an initialize method" unless init_defined?(presenter)
```

Comment:
@lukebergen you should be able to use `presenter.private_method_defined?(: initialize)` which takes into account super class methods. See: http://ruby-doc.org/core-2.6.3/Module.html#method-i-private_method_defined-3F

Reply:
Based on that documentation, it looks like that'd return false unless the `initialize` was defined as a private method. Which I don't think we'd want?

I wrote it the way I did because I don't want Object's initialize to "count" but _do_ want to allow for Object > GrandparentPresenter > ParentPresenter > ..... > VerySpecificPresenter to allow for anybody up the ancestor chain to define initialize BUT as soon as we hit Object, we know that initialize shouldn't be considered a legitimate Presenter initialize.

I didn't dig too deep into the libraries core though so I may be missing something.

Reply:
@kernow any thoughts on this?

Reply:
@lukebergen I was away for most of last week, just catching up now 🙂, I expected that `private_method_defined?` would not include `Object` but looks like thats not that case. Your solution looks good in that case. I'll get it merged and a gem released in the next couple of days. Thanks for your efforts!